community/values/src/main/java/org/neo4j/values/storable/TemporalValue.java

```diff
@@ -1366,6 +1367,26 @@ static int validNano( AnyValue millisecond, AnyValue microsecond, AnyValue nanos
         return (int) (ms * 1000_000 + us * 1000 + ns);
     }
 
+    static <TEMP extends Temporal> TEMP withTruncatedNano( TEMP value, int precision )
+    {
+        if ( precision < 0 || precision > 9 )
+        {
+            throw new InvalidValuesArgumentException( "The default_temporal_nanosecond_precision config must be an integer between 0 and 9, was " + precision );
+        }
+        int truncateFactor = (int) Math.pow( 10, 9 - precision );
```

Comment:
It might be an optimization to skip the algorithm and associated object creation if precision == 9.

Reply:
Good point!